feat: Add initial support for Blobscan

PabloCastellano opened the pull request.

barnabasbusa commented:
can you pls run `kurtosis lint --format`

barnabasbusa commented on src/package_io/input_parser.star:
```diff
@@ -45,6 +45,7 @@ DEFAULT_ADDITIONAL_SERVICES = [
     "el_forkmon",
     "beacon_metrics_gazer",
     "dora",
+    "blobscan",
```
I personally wouldn't enable this as default.

barnabasbusa commented on network_params.yaml:
```diff
@@ -31,7 +31,7 @@ participants:
   network_id: "3151908"
   deposit_contract_address: "0x4242424242424242424242424242424242424242"
-  seconds_per_slot: 12
+  seconds_per_slot: 4
```
why do you need to change these?

barnabasbusa commented on main.star:
```diff
@@ -328,14 +330,25 @@ def run(plan, args={}):
                 network_params.electra_fork_epoch,
             )
             plan.print("Successfully launched dora")
+            DORA_BASE_URL="http://{0}:{1}".format(dora_config.ip_address, dora_config.ports["http"].number)
```
why are you using DORA for all names ?

barnabasbusa commented:
Can you make sure your commits are signed?

PabloCastellano commented:
@barnabasbusa I have addressed your comments and removed unrelated changes. Commits are now squashed and signed.

barnabasbusa commented:
Could you please add some docs regarding blobscan in the readme? 
Also I see that network_params.yaml still has blobscan as a default tooling enabled, could you please remove that?

PabloCastellano commented:
@barnabasbusa I removed blobscan from `DEFAULT_ADDITIONAL_SERVICES` in `src/package_io/input_parser.star` and thought that was enough but ok I have now removed it also from `additional_services` from `network_params.yaml`

barnabasbusa commented:
Would blobscan require a dora instance? 

It looks like you are passing dora url to blobscan. Would blobscan be able to operate without dora? If blobscan won't work without dora, then might worth adding a check there somewhere.

PabloCastellano commented:
@barnabasbusa Blobscan can actually run without dora. It will simply not display the link to the beaconchain explorer. I will remove the dependency and add a comment about it

barnabasbusa commented:
Would actually be good if you would use dora if its available, and don't use it if it is not. Sorry for the possible misunderstanding. I think its good to have all functionality if Dora is available.

barnabasbusa commented:
Please add blobscan to readme line 265 too.

PabloCastellano commented on main.star:
```diff
@@ -328,14 +330,27 @@ def run(plan, args={}):
                 network_params.electra_fork_epoch,
             )
             plan.print("Successfully launched dora")
+            SERVICE_URLS["dora"] = "http://{0}:{1}".format(
```
This is not working as expected.

```
5d87b7d55815:/app$ env|grep NEXT_
NEXT_PUBLIC_EXPLORER_BASE_URL=http://172.16.0.15:8080
NEXT_PUBLIC_NETWORK_NAME=kurtosis-devnet
NEXT_PUBLIC_BEACON_BASE_URL=http://172.16.0.15:8080
```

Dora listens internally at :8080 but exposes 33154:

```
5c5046128288   dora      http: 8080/tcp -> http://127.0.0.1:33154       RUNNING
```

Is it possible to retrieve the external address from dora_config?

PabloCastellano commented on main.star:
```diff
@@ -328,14 +330,27 @@ def run(plan, args={}):
             plan.print("Successfully launched dora")
+            SERVICE_URLS["dora"] = "http://{0}:{1}".format(
+                dora_config.ip_address, dora_config.ports["http"].number
+            )
```
This is not working as expected.

```
5d87b7d55815:/app$ env|grep NEXT_
NEXT_PUBLIC_EXPLORER_BASE_URL=http://172.16.0.15:8080
NEXT_PUBLIC_NETWORK_NAME=kurtosis-devnet
NEXT_PUBLIC_BEACON_BASE_URL=http://172.16.0.15:8080
```

Dora listens internally at :8080 but exposes 33154:

```
5c5046128288   dora      http: 8080/tcp -> http://127.0.0.1:33154       RUNNING
```

Is it possible to retrieve the external address from dora_config?

tedim52 replied:
Hey @PabloCastellano ! Actually this is expected. The dora explorer and the blobscan services are running within the same isolated docker network created by the enclave so the blobscanner will be able to access dora the explorer on port 8080. 

Port 33154 is the port that a service running on your host docker network can use to get to dora's 8080 port!

PabloCastellano replied:
@tedim52 That won't work because the variables expect an external URL. This way links can be clicked in blobscan and opened in dora. If the internal isolated address:port is used, the link will be broken.

As example, check out https://blobscan.com/block/195071 and click on the Slot number. You are being redirected to dora n because `NEXT_PUBLIC_BEACON_BASE_URL` is set `https://dora.dencun-devnet-11.ethpandaops.io/`

PabloCastellano replied:
Ok, so I am going to remove this functionality. Can you share the link to the issue so I can link it from a comment?

PabloCastellano commented on src/cl/teku/teku_launcher.star:
```diff
@@ -29,8 +29,8 @@ METRICS_PORT_NUM = 8008
 BEACON_MAX_CPU = 1000
-BEACON_MIN_MEMORY = 512
-BEACON_MAX_MEMORY = 1024
+BEACON_MIN_MEMORY = 1024
```
@barnabasbusa found that teku was crashing due to OOM and fixed it in the same commit

h4ck3rk3y commented on src/blobscan/blobscan_launcher.star:
```diff
@@ -0,0 +1,124 @@
+        el_client_contexts[0].ip_addr, el_client_contexts[0].rpc_port_num
+    )
+
+    postgres_output = postgres.run(plan)
```
can you pass in `service_name = "blobscan-postgres"`

h4ck3rk3y replied:
```suggestion
    postgres_output = postgres.run(plan, service_name="blobscan-postgres")
```

h4ck3rk3y commented on main.star:
```diff
@@ -21,6 +21,7 @@ beacon_metrics_gazer = import_module(
     "./src/beacon_metrics_gazer/beacon_metrics_gazer_launcher.star"
 )
 dora = import_module("./src/dora/dora_launcher.star")
+blobscan = import_module("./src/blobscan/blobscan_launcher.star")
```
@PabloCastellano can you add a testcase in `.github/tests` that spins up blobscan! there is a tools example; it might be as simple as adding a line there

h4ck3rk3y replied:
https://github.com/kurtosis-tech/ethereum-package/blob/main/.github/tests/mix-with-tools.yaml - just enrich this one;
